test/jit/test_device_analysis.py

```diff
@@ -75,6 +112,29 @@ def add_self(x):
         # self.prop_device_on_graph(graph, [self.cpu])
         self.assertEqual(graph_input.type().device(), self.cpu)
 
+    @unittest.skipIf(models is None, "Requires torchvision")
+    def test_mobilenet(self):
+        in_cpu = torch.randn(1, 3, 224, 224, device=self.cpu)
+        in_example = in_cpu
+
+        expected_device = self.cpu
+        m = torch.jit.script(models.mobilenet_v3_small())
+        m.eval()
+        graph = torch.jit.freeze(m).graph
+        # torch._C._jit_pass_erase_shape_information(graph)
+        apply_input_props_using_example(graph, in_example)
+        torch._C._jit_pass_propagate_shapes_on_graph(graph)
+        torch._C._jit_pass_propagate_device(graph)
+
+        actual_device = self.node_output_device(graph)
+
+        if expected_device is None or actual_device is None:
```

Comment:
this check seems to be more complicated than it needs to be.
`expected_device` can't be None since you are assigning it a few lines above and never overwrite it. 
`self.assertEqual(actual_device, expected_device)` should just work but it seems you want to separate a case where actual_device was infered to some device but it's not equal to the expected one ?